src/node_state/ordered_tx_map.rs

```diff
@@ -91,6 +95,22 @@ impl OrderedTxMap {
                 }
             } && is_next;
         }
+        for contract in extra_contracts_to_block {
+            is_next = match self.tx_order.get_mut(&contract) {
+                Some(vec) => {
+                    vec.push_back(tx.hash.clone());
+                    vec.len() == 1
+                }
+                None => {
+                    self.tx_order.insert(contract.clone(), {
+                        let mut vec = VecDeque::new();
+                        vec.push_back(tx.hash.clone());
+                        vec
+                    });
+                    true
+                }
```

Comment:
duplicated code, would be great to simplify

Reply:
Marche pas pour une raison un peu bresom de "je vais me retrouver à enlever deux fois la TX de la liste" en cas de delete_self, je pense que je vais revoir un peu la logique pour éviter la situation.